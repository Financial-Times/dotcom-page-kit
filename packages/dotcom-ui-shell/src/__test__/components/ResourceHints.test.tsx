import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../../components/ResourceHints'

describe('dotcom-ui-shell/src/components/ResourceHints', () => {
  it('renders the given paths and URLs as resource hints', () => {
    const fixture = [
      'www.example.com/assets/style.css',
      'www.example.com/images/graphic.svg#icon',
      '/assets/public/style.as83hd99.css',
      '/__origami/service/build/v3/font?font_format=woff2&font_name=FinancierDisplayWeb-Bold&system_code=page-kit&version=1.12'
    ]

    const tree = renderer.create(<Subject resourceHints={fixture} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
