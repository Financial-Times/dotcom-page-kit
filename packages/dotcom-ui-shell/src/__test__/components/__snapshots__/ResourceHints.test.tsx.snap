// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-shell/src/components/ResourceHints renders the given paths and URLs as resource hints 1`] = `
Array [
  <link
    href="https://spoor-api.ft.com"
    rel="preconnect"
  />,
  <link
    crossOrigin="use-credentials"
    href="https://session-next.ft.com"
    rel="preconnect"
  />,
  <link
    href="https://ads-api.ft.com"
    rel="preconnect"
  />,
  <link
    href="https://securepubads.g.doubleclick.net"
    rel="preconnect"
  />,
  <link
    as="style"
    href="www.example.com/assets/style.css"
    rel="preload"
    type="text/css"
  />,
  <link
    as="image"
    href="www.example.com/images/graphic.svg#icon"
    rel="preload"
    type={null}
  />,
  <link
    as="script"
    href="http://polyfill.io/v3/bundle.min.js?features=es5,es6"
    rel="preload"
    type={null}
  />,
  <link
    as="style"
    href="/assets/public/style.as83hd99.css"
    rel="preload"
    type="text/css"
  />,
  <link
    as="font"
    crossOrigin="anonymous"
    href="/__origami/service/build/v2/files/o-fonts-assets/FontFace.woff"
    rel="preload"
    type="font/woff"
  />,
]
`;
